### PR TITLE
manual deployment kubernetes document on ubuntu

### PR DESCRIPTION
**Pull request by MengZn**
The problem is this manual deployment kubernetes document in ubuntu is too old.In addition, the ubuntu manual installation guide has been deleted.

So I made a solution that manually deployment kubernetes on ubuntu.
This program has been tested and experimented many times, test and experimental results show that this is a good deploy solution.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/5431)

### CODE REVIEW

**Comment by MengZn** [comment]
Reviewed 23 of 25 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/5431)*

**Comment by MengZn** [comment]
cc @zhangxiaoyu-zidif

**Comment by zhangxiaoyu-zidif** [comment]
/cc @kairen
do you have interest to review it? I see you removed some outdated ubuntu deployment files.

File: docs/getting-started-guides/ubuntu/file/etcd/etcd.conf [hunk 1]
```diff
@@ -0,0 +1,10 @@
+ETCD_NAME=master
+ETCD_DATA_DIR=/var/lib/etcd
+ETCD_INITIAL_ADVERTISE_PEER_URLS=http://192.168.16.100:2380
+ETCD_INITIAL_CLUSTER=master=http://192.168.16.100:2380
```
Comment (zhangxiaoyu-zidif) [inline docs/getting-started-guides/ubuntu/file/etcd/etcd.conf]: could you use hostname to replace the specific IP and add a host file to map hostnames and specific IPs
it may confuse elementary user.
e.g. 192.168.16.100 -> master.example.com

File: docs/getting-started-guides/ubuntu/file/flannel/flanneld [hunk 2]
```diff
@@ -0,0 +1,3 @@
+FLANNEL_ETCD_ENDPOINTS="http://192.168.16.100:2379"
```
Comment (zhangxiaoyu-zidif) [inline docs/getting-started-guides/ubuntu/file/flannel/flanneld]: specific IP -> hostname

File: docs/getting-started-guides/ubuntu/file/flannel/flanneld [hunk 3]
```diff
@@ -0,0 +1,3 @@
...
+FLANNEL_ETCD_PREFIX="/atomic.io/network"
+FLANNEL_OPTIONS="--iface=enp0s8"
```
Comment (zhangxiaoyu-zidif) [inline docs/getting-started-guides/ubuntu/file/flannel/flanneld]: *enp0s8* may be a specific parameter. So it is not suitable to list it here. I think.

**Comment by steveperry-53** [comment]
cc @ktsakalozos @marcoceppi @castrojo

**Comment by steveperry-53** [comment]
@MengZn @zhangxiaoyu-zidif I just took a quick look at this. I can do an edit pass, but I think I'd like some others to comment on the big picture of how this fits in to the overall set of Ubuntu docs.

@ktsakalozos @marcoceppi @castrojo Can you comment? Or can you suggest others who should comment on this. Thanks.

**Comment by marcoceppi** [comment]
Thank you @MengZn for taking time to write something like this out! I really appreciate people contributing to the Kubernetes documentation especially when it's for Ubuntu :smile: . However, I'm concerned at how raw, in general, this document is. I can see value in doing a fully manual deploy of Kubernetes from straight raw bits as a means to deconstruct how Kubernetes operates. However, I don't see the value in pushing this as a form of documenting for a user to follow.

For a manual deployment of Kubernetes on Ubuntu I would expect instructions to revolve around either `kubeadm` or potentially the snaps, otherwise this is simply [Kubernetes the hard way](https://github.com/kelseyhightower/kubernetes-the-hard-way) but as a set of instructions.

That said, I don't think it's a bad thing to have, I'm just going to take some time to figure out how to best fit and include this documentation - unless others have ideas.

**Comment by castrojo** [comment]
cc @kubernetes/sig-cluster-lifecycle-misc 

I think there is value in having directions like this somewhere, though I also have concerns on how it should be presented in the official documentation, if at all. 

For example, Kubernetes uses an older version of Docker that we package and distribute that we know has test coverage, whereas this grabs pure upstream versions of Docker and etcd. How do people do upgrades, etc? RBAC? 

I think if we want "the hard way" style instructions in the docs that they should have a proper warning/expectations set at the top of the page so that people don't assume that it's an officially supported configuration and then get burnt by say, an upstream docker bug from a version that isn't being tested.

**Comment by steveperry-53** [comment]
@MengZn @zhangxiaoyu-zidif @castrojo @marcoceppi @heckj @luxas What's the action item here? Do you think we should put this in the docs at kubernetes.io (with a big warning)? Or should this be published somewhere outside of kubernetes.io?

cc @chenopis @zacharysarah @kubernetes/sig-docs-maintainers

**Comment by zhangxiaoyu-zidif** [comment]
@steveperry-53 
Maybe this PR is valuable, but not proper in kubernetes.io.
I think it's better to close this PR and if there is necessary then publish it somewhere else...

**Comment by xiangpengzhao** [comment]
I remember @resouer 's SEL team once maintained a similar solution in kubernetes.io, but that solution was finally deprecated and removed. @resouer any idea or suggestion here? Thanks!

**Comment by zhangxiaoyu-zidif** [comment]
@xiangpengzhao 
is this PR(https://github.com/kubernetes/website/pull/5431) you mentioned?

**Comment by xiangpengzhao** [comment]
@zhangxiaoyu-zidif exactly.

**Comment by chenopis** [comment]
So what do we want to do with this PR?

**Comment by MengZn** [comment]
@chenopis @steveperry-53 @marcoceppi I think this can be provided to those who want to try to manually install .
This file does not represent the latest version of the installation, but you can try to deploy it manually. 
I think it can be placed in one of the ways in ubuntu installs manually

**Comment by luxas** [comment]
FWIW; sig-cluster-lifecycle is working towards simplifying and unifying the cluster creation/bootstrapping process. We want to keep things as simple as possible for end users. kubeadm is the tool the community develops for this purpose, and can then be used by even higher-level tools like kops (that manage a full cluster's lifecycle vs kubeadm that manages one master or node at a time).

As much as I hate saying no to good work, I must do so in this case. We actively are trying to delete old deployment solutions in this repo; which is exactly the opposite to what this PR does.

TL;DR; the problems with these getting started guides are:
 - They normally are written/maintained by one or two persons; not a full SIG. This leads to SPOF and outdated docs.
 - It is immensely hard to maintain such a guide. Trust me, I've done this (and failed to do so) myself earlier. Kubernetes changes in a insane pace; and duplicating manifests (like kube-dns copied over here) is sub-optimal. For example, whose responsibility is it to update this manifest whenever kube-dns changes? The sig-network team or the maintainers of this guide? The answer is the maintainers of this guide; which means it will eventually get undone.

Some quick feedback on this guide:
 - This doc favors flannel directly. We are really trying as a community to not merge such documentation on the vendor-neutral http://kubernetes.io/docs site.
 - etcd here is unsecured, which means anyone with ping access to your master have root access to your cluster. The same goes for many other places; like the kubelet API that is unsecured.
 - A today necessary component in a good cluster is RBAC enabled by default. This is missing.
 - There are no guides how to upgrade your cluster, and the manifests referenced here aren't the latest version (and will never be updated frequently enough).

So, to wrap up: This is good work @MengZn, thank you for submitting! But it will not, and should not, live in this place. Users can create clusters with kubeadm just fine, on Ubuntu as well as a lot of other OSes.

We could definitely use some help with documenting how the cluster is set up when using kubeadm though. I'd be delighted to see your help there instead, explaining what kubeadm does for you and how you then can extend it to your liking if you want.

If you're interested in helping the community, please attend the SIG Cluster Lifecycle meeting on Tuesday 9am PT. Again, thank you for your initiative!
